### ISSUE TITLE
Recursively search scripts with "shellcheck -x".

### ISSUE THREAD

**Issue by orbea** [koalaman/shellcheck#1034]
#### For new checks and feature suggestions
- [x] shellcheck.net (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh

#!/bin/sh

foo=bar

. foo.sh
. bar.sh

echo "$bar"
```

#### Here's what shellcheck currently says:

```
Line 3:
foo=bar
^-- SC2034: foo appears unused. Verify it or export it.
 
Line 5:
. foo.sh
^-- SC1091: Not following: foo.sh was not specified as input (see shellcheck -x).
 
Line 6:
. bar.sh
^-- SC1091: Not following: bar.sh was not specified as input (see shellcheck -x).
 
Line 8:
echo "$bar"
      ^-- SC2154: bar is referenced but not assigned.
```

#### Here's what I wanted or expected to see:

It is possible that `$foo` is being used and that `$bar` is being set in either `./foo.sh` or `./bar.sh`, but shellcheck does not recursively look into these files and print warnings for them. It is understandable to see these warnings at https://www.shellcheck.net/, but when a user is using a local copy of shellcheck with `-x` it should be possible even if as an optional feature to recursively search these scripts as `./foo.sh` may not source any files itself and its warnings may not be correct as a result.

The alternative is to compile these scripts into a single file while may not fit the specific project and is time consuming for mere purpose of testing.

**Comment by orbea** [comment]
To clarify I mean that I want it print warnings for `./foo.sh` and `./bar.sh` along with the main script in this example.

**Comment by Pawamoy** [comment]
Imagine you run shellcheck on several scripts, and each of them are sourcing one or several other files (the same ones). How do you handle that? Because if shellcheck "follows" the sourced files and checks them, you will have duplicated warnings, and the output will be very noisy.

Anyway, if this is not the standard behavior, and has to be activated with an option, I'm not against it.

**Comment by orbea** [comment]
Yea, I agree it should not be the default behavior, but without being able to test all related scripts in the same scope you will get many noisy false warnings about `$var is referenced but not assigned.` or `$var appears unused. Verify it or export it.`.

**Comment by Pawamoy** [comment]
By the way, you know you can specify the source in the script with comments like `# shellcheck source=foo.sh` right?

**Comment by orbea** [comment]
I did not know, is that different from using `shellcheck -x'`?

**Comment by Pawamoy** [comment]
I think the `-x` tells shellcheck not to trigger warning if you source a file different than the ones specified on the command-line. To actually do what you want, you should use both `-x` and `source` directives: https://github.com/koalaman/shellcheck/wiki/Directive

**Comment by orbea** [comment]
Thanks for the explanation, but that doesn't quite do what I want. The goal is to also print warnings found in `foo.sh` and `bar.sh` along with any other scripts they source while in the scope of the main script.

**Comment by orbea** [comment]
Yes, `shellcheck -ax` seems to resolve this! Thanks for pointing it out.
